test/albumController.test.js

```diff
@@ -0,0 +1,105 @@
+const { app, connectDB } = require('../src/expressServer'); // Import your Express app here
+const { StatusCodes } = require('http-status-codes');
+const request = require('supertest');
+const { MongoMemoryServer } = require('mongodb-memory-server');
+const Album = require('../src/models/Album');
+
+const {
+  getAllAlbums,
+  getSingleAlbum,
+  getAlbumWithAllUsersWhoPurchasedIt,
+  getFilteredAlbums,
+} = require('../src/controllers/albumController');
+
+let server;
+let mongooseConnection;
+let mongodb;
+
+beforeAll(async () => {
+  mongodb = await MongoMemoryServer.create();
+  const url = mongodb.getUri();
+  process.env.MONGO_URL = url;
+  mongooseConnection = await connectDB(url);
+  server = await app.listen(8001);
+});
+
+afterAll(async () => {
+  await server.close();
+  await mongooseConnection.disconnect();
+  await mongodb.stop();
+});
+
+describe('AlbumController API Tests', () => {
+  const mockAlbumData = {
+    albumName: 'Test Album',
+    artistName: 'Test Artist',
+    price: 9.99,
+    spotifyUrl: 'https://api.spotify.com/v1/albums/',
+  };
+
+  beforeEach(async () => {
+    await Album.deleteMany({});
+    await Album.create(mockAlbumData);
+  });
+
+  it('should test the getAllAlbums endpoint - Success Case', async () => {
+    const response = await request(app).get('/api/v1/albums');
+    expect(response.status).toBe(StatusCodes.OK);
+    expect(response.body).toHaveProperty('albums');
+    expect(response.body.albums).toHaveLength(1);
+  });
+
+  it('should test the getAllAlbums endpoint - Error Case (Unauthorized)', async () => {
+    const response = await request(app).get('/api/v1/albums');
+    expect(response.status).toBe(StatusCodes.UNAUTHORIZED);
+  });
+
+  it('should test the getSingleAlbum endpoint - Success Case', async () => {
+    const albums = await Album.find({});
+    const albumId = albums[0]._id;
+
+    const response = await request(app).get(`/api/albums/v1/${albumId}`);
+    expect(response.status).toBe(StatusCodes.OK);
+    expect(response.body).toHaveProperty('album');
+  });
+
+  it('should test the getSingleAlbum endpoint - Error Case (Not Found)', async () => {
+    const response = await request(app).get('/api/albums/v1/invalidAlbumId');
+    expect(response.status).toBe(StatusCodes.NOT_FOUND);
+  });
+
+  it('should test the getAlbumWithAllUsersWhoPurchasedIt endpoint - Success Case', async () => {
+    const albums = await Album.find({});
+    const albumId = albums[0]._id;
+
+    const response = await request(app).get(
+      `/api/albums/${albumId}/purchased-users`
+    );
+    expect(response.status).toBe(StatusCodes.OK);
+    expect(response.body).toHaveProperty('album');
+    expect(response.body).toHaveProperty('purchasingUsersCount');
+  });
+
+  it('should test the getAlbumWithAllUsersWhoPurchasedIt endpoint - Error Case (Not Found)', async () => {
+    const response = await request(app).get(
+      '/api/albums/invalidAlbumId/purchased-users'
```

Comment:
```suggestion
      '/api/v1/albums/invalidAlbumId/listOfUsersWhoPurchasedThisAlbum'
```

^ correct URL (with invalid album id)